src/trans.c

```diff
@@ -3151,10 +3151,11 @@ Obj FuncCYCLES_TRANS(Obj self, Obj f)
         return out;
     }
 
-    seen = ResizeInitTmpTrans(deg);
     out = NEW_PLIST(T_PLIST, 0);
     nr = 0;
 
+    seen = ResizeInitTmpTrans(deg);
+
```

Comment:
Argh, damn, we went through that code so often, looking for exactly these kinds of problems, and still missed this one ... :-(